src/_gettsim/transfers/benefit_checks/benefit_checks.py

```diff
@@ -1,97 +1,245 @@
+from _gettsim.shared import policy_info
+
 aggregate_by_group_benefit_checks = {
-    "wohngeld_vorrang_wthh": {
-        "source_col": "wohngeld_vorrang_bg",
-        "aggr": "any",
-    },
-    "wohngeld_kinderzuschl_vorrang_wthh": {
-        "source_col": "wohngeld_kinderzuschl_vorrang_bg",
-        "aggr": "any",
+    "alle_wohngeld_kinderzuschl_statt_arbeitsl_geld_2_fg": {
+        "source_col": "wohngeld_kinderzuschl_statt_arbeitsl_geld_2_endogen",
+        "aggr": "all",
     },
 }
 
 
-def wohngeld_vorrang_bg(
+def wohngeld_kinderzuschl_statt_arbeitsl_geld_2_endogen(
+    wohngeld_kinderzuschl_vorrangig_bg: bool,
+    wohngeld_kinderzuschl_günstiger: bool,
+) -> bool:
+    """Individual receives Wohngeld and Kinderzuschlag instead of Arbeitslosengeld II.
+
+    Calculated endogenously.
+
+    Parameters
+    ----------
+    wohngeld_vorrangig_bg
+        See :func:`wohngeld_vorrangig_bg`.
+    wohngeld_kinderzuschl_günstiger
+        See :func:`wohngeld_kinderzuschl_günstiger`.
+
+    Returns
+    -------
+
+    """
+    return wohngeld_kinderzuschl_vorrangig_bg or wohngeld_kinderzuschl_günstiger
+
+
+def wohngeld_kinderzuschl_statt_arbeitsl_geld_2(
```

Comment:
We need to be really careful with the names here. It seems like this one applies to children only, but there is nothing in the name suggesting that.

Also, I am not sure we need / want the `wohngeld_kinderzuschl_statt_arbeitsl_geld_2_endogen` / `wohngeld_kinderzuschl_statt_arbeitsl_geld_2` distinction. Maybe that even applies to `bg_id` and `wthh_id` and their endogenous counterparts.

I would think we'll need two rounds of calling `compute_taxes_and_transfers`:
1. Calculate `bg_id`, `wthh_id`, `wohngeld_kinderzuschl_statt_arbeitsl_geld_2`
2. Given these three, we can just calculate `arbeitsl_geld_2`, `wohngeld`, `kinderzuschlag` without any checks on which one applies.

Trying to do both in one go would raise some circular graph error. I might be missing something (again), though.

Reply:
Just a comment on this: The variable is not relevant for children only, it also determines the type of transfer for the adults.

If it is set endogenously to create candidates, it always results in `False` for the adults in the FG (hence your comment probably)

Reply:
Exactly. The code needs that as a  description.